plasoscaffolder/frontend/frontend.py

```diff
@@ -0,0 +1,373 @@
+# !/usr/bin/python
+# -*- coding: utf-8 -*-
+"""The scaffolder frontend."""
+import re
+
+from typing import Dict
+from typing import List
+from typing import Type
+
+from plasoscaffolder.frontend import output_handler
+from plasoscaffolder.helpers import git
+from plasoscaffolder.lib import engine
+from plasoscaffolder.lib import errors
+
+from plasoscaffolder.definitions import interface as definition_interface
+from plasoscaffolder.definitions import manager as definition_manager
+from plasoscaffolder.scaffolders import interface as scaffolder_interface
+from plasoscaffolder.scaffolders import manager as scaffolder_manager
+
+
+class ScaffolderFrontend:
+  """A frontend implementation for the scaffolder project.
+
+  Attributes:
+    OUTPUT_HANDLER (output_handler.BaseOutputHandler: output handler
+        that is used to request and read input from end user.
+  """
+
+  OUTPUT_HANDLER = output_handler.BaseOutputHandler()
+
+  @classmethod
+  def _AskDictQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> Dict[str, Type]:
+    """Ask the user a question and return a dict answer back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      dict: the answer as supplied by the user.
```

Comment:
Update docstring to describe the content of the dict. It doesn't seem to return an Answer class or anything...

Reply:
I think that'll be overkill. Added something here.